libs/partners/anthropic/langchain_anthropic/experimental.py

```diff
@@ -118,21 +128,44 @@ def _xml_to_dict(t: Any) -> Union[str, Dict[str, Any]]:
     return d
 
 
-def _xml_to_tool_calls(elem: Any) -> List[Dict[str, Any]]:
+def _xml_to_function_call(invoke: Any, tools: List[Dict]) -> Dict[str, Any]:
+    name = invoke.find("tool_name").text
+    arguments = _xml_to_dict(invoke.find("parameters"))
+
+    # make list elements in arguments actually lists
+    filtered_tools = [tool for tool in tools if tool["name"] == name]
+    if len(filtered_tools) > 0 and not isinstance(arguments, str):
+        tool = filtered_tools[0]
+        for key, value in arguments.items():
+            if key in tool["parameters"]["properties"]:
+                if "type" in tool["parameters"]["properties"][key]:
+                    if tool["parameters"]["properties"][key][
+                        "type"
+                    ] == "array" and not isinstance(value, list):
+                        arguments[key] = [value]
+                    if (
+                        tool["parameters"]["properties"][key]["type"] != "object"
+                        and isinstance(value, dict)
+                        and len(value.keys()) == 1
+                    ):
+                        arguments[key] = list(value.values())[0]
```

Comment:
crawl for objects like `{"item": "my string"}` that should really just be `"my string"`